plugins/org.jkiss.dbeaver.ext.hana/src/org/jkiss/dbeaver/ext/hana/model/HANADataSource.java

```diff
@@ -198,6 +203,32 @@ private boolean changeExpiredPassword(DBRProgressMonitor monitor, JDBCExecutionC
         return true;
     }
 
+    private boolean changeLicense(DBRProgressMonitor monitor, JDBCExecutionContext context, String purpose) {
+        DBPConnectionConfiguration connectionInfo = getContainer().getActualConnectionConfiguration();
+        String newLicense = DBWorkbench.getPlatformUI().promptText(
+                "HANA License - " + getContainer().getName(),"License invalid/expired, you can use this dialog to set a new license." + 
+                "\nWatch out for SYSDB vs Tenant in current connection as both can accept a license, however the SYSDB will overrule the Tenant license and is required." +
+                "\nWill attempt to set with current user: " + connectionInfo.getUserName() + " on Port: " + connectionInfo.getHostPort() + 
+                "\nIf afterwards the error persists, probably SYSDB license is still invalid or the new license is not valid.", "----- Begin SAP License -----\n...");
```

Comment:
I think it is worth localizing the title and the message. Even if you don't know any other languages besides English, it's still worth moving _potentially localizable_ pieces of text to message bundles.